setup.py

```diff
@@ -19,8 +19,22 @@
 from setuptools import find_packages, setup
 from setuptools.command.test import test as TestCommand
 from version import version
+from pkg_resources import parse_version
 
-toil_version = '3.1.4'
+# Toil version check -- Raise warning instead of using intall_requires to avoid virtualenv conflicts
+toil_minor_version = 3.1
```

Comment:
The augmented parse_version wouldn't simplify incrementing the minor version either. Just omit the arithmetic and hard-code both version limits in the code. That's still better than a brittle or overly complicated attempt at solving it programmatically.

```
toil_min_version='3.1.5'
toil_max_version='3.2'
```